packages/@lwc/synthetic-shadow/src/polyfills/mouse-event/polyfill.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { defineProperty, getOwnPropertyDescriptor, isNull } from '@lwc/shared';
+
+import { pathComposer } from '../../3rdparty/polymer/path-composer';
+import { retarget } from '../../3rdparty/polymer/retarget';
+import { eventCurrentTargetGetter } from '../../env/dom';
+import { isNodeShadowed } from '../../faux-shadow/node';
+import { getOwnerDocument } from '../../shared/utils';
+
+const relatedTargetGetter = getOwnPropertyDescriptor(MouseEvent.prototype, 'relatedTarget')!
+    .get as () => typeof MouseEvent.prototype.relatedTarget;
+
+defineProperty(MouseEvent.prototype, 'relatedTarget', {
```

Comment:
Is there any reason why not reusing the same code for both `MouseEvent` and `FocusEvent`. The only thing that changes is the prototype and the getter as far as I can tell.

Reply:
I'm usually ok with code repetition if it's simple and not repeated much (there are no other event interfaces using `relatedTarget`). We can refactor this if you think we should.

Reply:
Code size is not a concern to me, it's more about maintainability in the long run. I think it would be better to keep this logic centralized. For instance when fixing a bug in the `MouseEvent.prototype.relatedTarget`I wouldn't automatically think about porting the same fix to the `FocusEvent`.